client/controller.js

```diff
@@ -1,83 +1,76 @@
 
     // translate rate is 1000ms +/- 500ms (randomized)
     const translateRate = 1000 + Math.floor(((Math.random() - 0.5) * 500));
-    const newClient = new Client([...TEXTLIST], getWsAddr(), targetLang, translateRate);
+    const newClient = new Client([...textList], getWsAddr(elements), targetLang, translateRate);
     clients.push(newClient);
   }
 
   return clients;
 }
 
-function initLanguageSelector() {
-  for (const language of LANGUAGECODES) {
+function initLanguageSelector(elements) {
+  for (const language of languageCodes) {
     const optionEl = document.createElement('option');
     optionEl.value = language.code;
     optionEl.textContent = `${language.name} [${language.code}]`;
-    EL.languageSelector.appendChild(optionEl);
+    elements.languageSelector.appendChild(optionEl);
   }
 }
 
-function getWsAddr() {
-  return 'ws://' + EL.translateServerAddr.value;
+function getWsAddr(elements) {
+  return 'ws://' + elements.translateServerAddr.value;
 }
 
-function initElements() {
-  EL.textToTranslate = document.querySelector('#text-to-translate');
-  EL.translatedText = document.querySelector('#translated-text');
-  EL.languageSelector = document.querySelector('#language-selector');
-  EL.translateServerAddr = document.querySelector('#translate-server-address');
-  EL.serverAddressUpdate = document.querySelector('#server-address-update');
-  EL.numberOfSpeakers = document.querySelector('#number-of-speakers');
-  EL.startSimulation = document.querySelector('#start-simulation');
-}
+function addEventListeners(elements) {
+  const manualTranslator = new ManualTranslator(
+    elements.textToTranslate,
+    elements.translatedText,
+    elements.languageSelector,
+    getWsAddr(elements),
+  );
 
-function addEventListeners() {
-  EL.textToTranslate.addEventListener('keyup', () => {
-    MANUALTRANSLATOR.requestTranslation();
+  elements.textToTranslate.addEventListener('keyup', () => {
+    manualTranslator.requestTranslation();
   });
 
-  EL.languageSelector.addEventListener('change', () => {
-    MANUALTRANSLATOR.requestTranslation();
+  elements.languageSelector.addEventListener('change', () => {
+    manualTranslator.requestTranslation();
   });
 
-  EL.serverAddressUpdate.addEventListener('click', () => {
-    MANUALTRANSLATOR.updateWebSocket(getWsAddr());
+  elements.serverAddressUpdate.addEventListener('click', () => {
+    manualTranslator.updateWebSocket(getWsAddr());
   });
 
-  EL.startSimulation.addEventListener('click', runSimulation);
+  elements.startSimulation.addEventListener('click', () => {
+    runSimulation(elements);
+  });
 }
 
 function init() {
-  initElements();
-  initLanguageSelector();
-  EL.translateServerAddr.value = window.location.hostname + ':' + window.location.port;
-  MANUALTRANSLATOR = new ManualTranslator(
-    EL.textToTranslate,
-    EL.translatedText,
-    EL.languageSelector,
-    getWsAddr(),
-  );
-  addEventListeners();
+  const elements = new Elements();
```

Comment:
I think I can see what you did here. To avoid using a global variable, you've put it into the init function and then injected it as a parameter to all the functions called from it.

You don't need necessarily need to do this (it's not wrong per se). If I understand correctly, if you had `const elements = new Elements()` at the top level of the script that's imported as a "module", it shouldn't become available as a global variable (because, modules). That way you shouldn't need to inject it into all the functions as it should be available at the correct scope.

I could be wrong on the global variable front though. Try it, and see if you can access the `elements` variable in the chrome console or `window.elements` to be doubly sure. It should return `undefined` if all is well.